File: api/src/common/entity/episode.ts
import { Column, CreateDateColumn, Entity, ManyToOne, OneToMany, PrimaryGeneratedColumn } from 'typeorm';

import { Season } from './season';
import { WatchedEpisode } from './watched-episode';

@Entity()
export class Episode {
  @PrimaryGeneratedColumn()
  id: string;

  @Column({ nullable: true })
  providerId: string;

  @Column({ nullable: true })
  name?: string;

  @Column({ nullable: true })
  number?: number;

  @Column({ nullable: true })
  type?: string;

  @Column({ nullable: true })
  aired?: Date;

  @Column({ nullable: true })
  runtime?: number;

  @Column({ type: 'text', nullable: true })
  summary?: string;

  @ManyToOne(() => Season, season => season.episodes)
  season?: Season;

  @OneToMany(() => WatchedEpisode, watchedEpisode => watchedEpisode.episode)
  watches?: WatchedEpisode[];

  @CreateDateColumn()
  createdAt?: Date;
}

File: api/models/db/episode.go
package modelsdb

import (
	"time"
)

type Episode struct {
	Model
	ProviderID   *string          `json:"providerId"`
	Name         *string          `json:"name"`
	Number       *uint16          `json:"number" gorm:"uniqueIndex:idx_episode_uni"`
	Type         *string          `json:"type"`
	Aired        *time.Time       `json:"aired"`
	Runtime      *uint16          `json:"runtime"`
	Summary      *string          `json:"summary" gorm:"type:text"`
	SeasonID     uint             `json:"-" gorm:"uniqueIndex:idx_episode_uni"`
	Season       Season           `json:"-"`
	Watches      []WatchedEpisode `json:"watches"`
	Watched      bool             `gorm:"-" json:"watched"`
	SeasonNumber *uint8           `gorm:"-" json:"seasonNumber"`
}
